### PR TITLE
MRG: Cleaner use of tmin/tmax

### LINKED ISSUE #2645: BUG?: resample + concatenation breaks
I'm regularly confronted to an error when I resample a set of epochs, and try to concatenate them.

```
import numpy as np
from mne.epochs import concatenate_epochs  # BTW, why is concatenate_epochs not accessible from mne directly
nepoch, nchan, ntime = 100, 30, 20
# make dummy epochs1 and epoch2, sfreq = 250 Hz
epochs = concatenate_epochs([epochs1, epochs2])
# works

epochs1.resample(epochs1.info['sfreq'] // 2)
epochs2.resample(epochs2.info['sfreq'] // 2)
epochs = concatenate_epochs([epochs1, epochs2])
# error
```

```
RuntimeError                              Traceback (most recent call last)
<ipython-input-6-28041ffc5880> in <module>()
      4 epochs1.resample(125)
      5 epochs2.resample(125)
----> 6 epochs = concatenate_epochs([epochs1, epochs2])

/home/jrking/mne-python/mne/epochs.pyc in concatenate_epochs(epochs_list)
   2602     .. versionadded:: 0.9.0
   2603     """
-> 2604     return _finish_concat(*_concatenate_epochs(epochs_list))

/home/jrking/mne-python/mne/epochs.pyc in _finish_concat(info, data, events, event_id, tmin, tmax, baseline, selection, drop_log, verbose)
   2580                       baseline=baseline, add_eeg_ref=False,
   2581                       selection=selection, drop_log=drop_log,
-> 2582                       proj=False, on_missing='ignore', verbose=verbose)
   2583     out.drop_bad_epochs()
   2584     return out

/home/jrking/mne-python/mne/epochs.pyc in __init__(***failed resolving arguments***)
    280             if data.ndim != 3 or data.shape[2] != \
    281                     round((tmax - tmin) * self.info['sfreq']) + 1:
--> 282                 raise RuntimeError('bad data shape')
    283             self.preload = True
    284             self._data = data

RuntimeError: bad data shape
```

I think we already mentioned this error with @Eric89GXL , but I haven't found the issue.

### PR DESCRIPTION
**Pull request by Eric89GXL**
Closes #2645.

The bug was due to improperly setting `tmax`. I've changed `epochs` to derive it from `.times` instead, which should be a cleaner approach.

### CODE REVIEW

File: mne/epochs.py [hunk 1]
```diff
@@ -2437,32 +2438,15 @@ def bootstrap(epochs, random_state=None):
 
 def _check_merge_epochs(epochs_list):
     """Aux function"""
-    event_ids = set(tuple(epochs.event_id.items()) for epochs in epochs_list)
-    if len(event_ids) == 1:
-        event_id = dict(event_ids.pop())
-    else:
+    if len(set(tuple(epochs.event_id.items()) for epochs in epochs_list)) != 1:
         raise NotImplementedError("Epochs with unequal values for event_id")
```
Comment (Eric89GXL) [inline mne/epochs.py]: @dengemann you okay with these changes? I don't think the actual values for `event_ids`, `baseline`, `tmin`, and `tmax` need to be returned by the function. It's checking to make sure they're identical, so the values in the `epochs.copy()` below should already have the correct ones. Make sense?

Comment (dengemann) [inline mne/epochs.py]: yes fine with me.

**Comment by Eric89GXL** [comment]
Ready for review/merge from my end.

**Comment by jaeilepp** [comment]
LGTM

**Comment by agramfort** [comment]
so now if I do Epochs(tmin=-0.2) I don't have epochs.tmin == -0.2 ?

**Comment by Eric89GXL** [comment]
@agramfort you should have the same thing you had in `master`, which would be something close to `-0.2` based on `sfreq`.

**Comment by Eric89GXL** [comment]
In other words, the value of `epochs.tmin` shouldn't have changed, but how we store it interally is different. Instead of having the value in both `epochs.times[0]` and `epochs.tmin` and having to keep both up to date whenever we need to change one of them (and having bugs related to this), now we only keep the times in `epochs.times` and get `tmin` from `times[0]`. Make sense?

**Comment by agramfort** [comment]
so no API change?

if so and travis is happy +1 for merge

**Comment by Eric89GXL** [comment]
There should be no API change, correct

**Comment by agramfort** [comment]
ok got it

**Comment by Eric89GXL** [comment]
Cool, will merge once the CIs come back happy then

**Comment by Eric89GXL** [comment]
Thanks for the quick reviews

**Pull request merged**